src/MessageBird/Resources/Groups.php

```diff
@@ -73,9 +73,10 @@ public function getContacts(?string $id = null, ?array $parameters = [])
         if ($id === null) {
             throw new InvalidArgumentException('No group id provided.');
         }
+        $contactResource = new Contacts($this->httpClient);
```

Comment:
@MelvinLoos what's the reason for setting up a new Contacts instance here?

Reply:
The reason being that you are retrieving Contacts here (related to the Groups object). The original code would only change the `resourceName` variable and not the `object` variable. Thus it would do the correct API call but would parse the response to the Group object instead of the Contact object.

Another solution would have been to call the setObject method on $this after the setResourceName method but this feels messy because if you would then try to use another method of the Groups object it would parse to the now newly set class in the `object` variable (which would be Contact). 

Actually now that I think about it I would probably remove the setObject method entirely and hardcode the reference inside the class definition. Since this is basically a mapping between Resource class and the Object class. Why would you need to change this during render time?

Reply:
Sounds good to me. 👍